### PR TITLE
AP_PiccoloCAN: Remove duplicated code

### PR DESCRIPTION
**Pull request by amilcarlucas**
```
Binary Name      Text [B]     Data [B]     BSS (B)      Total Flash Change [B] (%)      Flash Free After PR (B)
---------------  -----------  -----------  -----------  ----------------------------  -------------------------
ardurover        0 (0.0000%)  0 (0.0000%)  0 (0.0000%)  0 (0.0000%)                                      336516
blimp            0 (0.0000%)  0 (0.0000%)  0 (0.0000%)  0 (0.0000%)                                      620952
arducopter       0 (0.0000%)  0 (0.0000%)  0 (0.0000%)  0 (0.0000%)                                      192324
arduplane        0 (0.0000%)  0 (0.0000%)  0 (0.0000%)  0 (0.0000%)                                      200432
ardusub          0 (0.0000%)  0 (0.0000%)  0 (0.0000%)  0 (0.0000%)                                      397880
antennatracker   0 (0.0000%)  0 (0.0000%)  0 (0.0000%)  0 (0.0000%)                                      643812
arducopter-heli  0 (0.0000%)  0 (0.0000%)  0 (0.0000%)  0 (0.0000%)                                      186124
```

### CODE REVIEW

**Comment by amilcarlucas** [comment]
This actually adds functionality and it costs 200bytes, not sure if it is worth it. I let the piccolo CAN maintainers decide

**Comment by peterbarker** [comment]
This one is conflicting badly now, but still seems to be relevant - the duplicate code is still present in PiccoloCAN.  We've had some recent activity in that directory - maybe we can drum up some interest.

@SchrodingersGat - this PR removes some code from PiccoloCAN by leveraging the ESC Telemetry library, and there's a few advantages in doing that - chiefly "less code is good code"

**Comment by SchrodingersGat** [comment]
@peterbarker do you refer to removing the `send_esc_telemetry_mavlink` function? It appears it is not being used anywhere. I'm happy to submit a PR which removes that function. Anything else which needs clean-up?

**Comment by amilcarlucas** [comment]
Rebased, please review again.

**Comment by magicrub** [comment]
What's with all the ! and <! stuff?

**Comment by amilcarlucas** [comment]
Half the code had Doxygen comments, half did not.
Now all code has doxygen comments.

**Comment by amilcarlucas** [comment]
@SchrodingersGat can you take it from here?

**Comment by amilcarlucas** [comment]
@SchrodingersGat is this going to bitrot again?

**Comment by SchrodingersGat** [comment]
@amilcarlucas I'll check it out

**Comment by amilcarlucas** [comment]
This will probably conflict with https://github.com/ArduPilot/ardupilot/pull/26252 so you better do it before that one gets merged.

File: libraries/AP_PiccoloCAN/AP_PiccoloCAN_Servo.cpp
Comment (reilly-callaway) [inline libraries/AP_PiccoloCAN/AP_PiccoloCAN_Servo.cpp:48]: Looks like this addition is breaking the compile currently.

This change shouldn't be needed as:
 - Servo data is already being logged in AP_PiccoloCAN.cpp
 - Currently other CAN servos (DroneCAN, etc.) aren't reporting voltage/current/temp/etc. over the ESC telem backend for mavlink anyway

**Review by reilly-callaway — CHANGES_REQUESTED** [review]

**Comment by SchrodingersGat** [comment]
@amilcarlucas we have reviewed and tested locally (after fixing compiling issues). With the changes suggested by @reilly-callaway above, we would be happy with this.

File: libraries/AP_PiccoloCAN/AP_PiccoloCAN_Servo.cpp
Comment (amilcarlucas) [reply libraries/AP_PiccoloCAN/AP_PiccoloCAN_Servo.cpp]: Thanks, fixed

**Comment by amilcarlucas** [comment]
Please approve so that this can get merged once the tests pass.

**Comment by amilcarlucas** [comment]
@peterbarker are you OK with the current changes?

**Comment by amilcarlucas** [comment]
Can someone re-trigger the failing sub test?

**Comment by peterbarker** [comment]
Merged, thanks!

**Comment by SchrodingersGat** [comment]
@amilcarlucas thanks for the work on this

**Pull request merged**
